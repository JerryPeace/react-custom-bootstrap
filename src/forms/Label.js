import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import styleMaps from '../styleMaps';

class Label extends React.Component {
    static propTypes = {
        text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
        labelAlign: PropTypes.oneOf(['left', 'right'])
    };

    static defaultProps = {
        inline: false,
        labelAlign: 'right'
    };

    renderLabelBase(children) {
        const {className, ...props} = this.props;

        return (
            <label className={classNames(className)}>{children}</label>
        );
    }

    render() {
        const props = this.props;
        let text = props.text,
            labelAlign = props.labelAlign;

        if (labelAlign === 'left') {
            return this.renderLabelBase([
                text,
                props.children
            ]);
        } else {
            return this.renderLabelBase([
                props.children,
                text
            ]);
        }
    }
}

export default Label;
