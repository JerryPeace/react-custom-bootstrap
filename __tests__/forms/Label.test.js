import React from 'react';
import {mount} from 'enzyme';
import toJSON from'enzyme-to-json';
import Label from 'forms/Label';

describe('Label', () => {
    let label;

    test('with default settings', () => {
        label = mount(<Label text='label text'>children</Label>);
        expect(toJSON(label)).toMatchSnapshot();
    });

    test('with empty text', () => {
        label = mount(<Label text=''></Label>);
        expect(toJSON(label)).toMatchSnapshot();
    });

    test('with labelAlign is left', () => {
        label = mount(<Label text='label text' labelAlign='left'>children</Label>);
        expect(toJSON(label)).toMatchSnapshot();
    });
});
