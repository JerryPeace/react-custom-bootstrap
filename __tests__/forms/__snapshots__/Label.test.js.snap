// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label with default settings 1`] = `
<Label
  inline={false}
  labelAlign="right"
  text="label text"
>
  <label
    className=""
  >
    children
    label text
  </label>
</Label>
`;

exports[`Label with empty text 1`] = `
<Label
  inline={false}
  labelAlign="right"
  text=""
>
  <label
    className=""
  />
</Label>
`;

exports[`Label with labelAlign is left 1`] = `
<Label
  inline={false}
  labelAlign="left"
  text="label text"
>
  <label
    className=""
  >
    label text
    children
  </label>
</Label>
`;
